src/main/scala/is/hail/table/Table.scala

```diff
@@ -82,15 +82,22 @@ object Table {
     new Table(hc, TableRead(path, spec, dropRows = false))
   }
 
-  def parallelize(hc: HailContext, rows: java.util.ArrayList[Row], signature: TStruct,
+  def parallelize(hc: HailContext, rowsJSON: String, signature: TStruct,
     keyNames: java.util.ArrayList[String], nPartitions: Option[Int]): Table = {
-    parallelize(hc, rows.asScala.toArray, signature, keyNames.asScala.toArray, nPartitions)
+    parallelize(hc, rowsJSON, signature, keyNames.asScala.toArray, nPartitions)
   }
 
-  def parallelize(hc: HailContext, rows: IndexedSeq[Row], signature: TStruct,
+  def parallelize(hc: HailContext, rowsJSON: String, signature: TStruct,
     key: IndexedSeq[String], nPartitions: Option[Int] = None): Table = {
+    val parsedRows = JSONAnnotationImpex.importAnnotation(JsonMethods.parse(rowsJSON), TArray(signature))
+      .asInstanceOf[IndexedSeq[Row]]
+    parallelize(hc, parsedRows, signature, key, nPartitions)
+  }
+
+  def parallelize(hc: HailContext, rows: IndexedSeq[Row], signature: TStruct,
+    key: IndexedSeq[String], nPartitions: Option[Int]): Table = {
     val typ = TableType(signature, key.toArray.toFastIndexedSeq, TStruct())
-    new Table(hc, TableParallelize(typ, rows, nPartitions))
+    new Table(hc, TableParallelize(typ, rows.asInstanceOf[IndexedSeq[Row]], nPartitions))
```

Comment:
Why do you need this cast here? doesn't rows already have to be `IndexedSeq[Row]`?

Reply:
yep, looks like a refactoring mistake